replace_substring_in_filenames.sh

```diff
@@ -17,6 +17,7 @@
 # Check if arguments are provided. If no arguments given, display user prompts 
 # of what to enter.
 if [ $# -eq 0 ]; then
+  IFS=''
   echo "Type the directory path for filenames that need changes: "
   read directory_path
   
```

Comment:
It's not super-important for your script, but when you change a variable like `IFS` that can affect the behaviour of the shell or its built-in commands like `read`, it's a good idea to save its original value and then restore that original value when you don't need the special behaviour. So you'd do something like

```
  OLD_IFS=$IFS
  IFS=''
  echo "Type the directory path for filenames that need changes: "
  read directory_path

  ... # other places where you call 'read' 
  read replacement_string
  IFS=$OLD_IFS
```

Reply:
Now when you're done with the `read` calls where you need `IFS` to be empty, you restore `IFS` to its original value.

Reply:
@kcharter As you suggested, I added this safeguard and re-tested.